src/Ocelot.Provider.Kubernetes/KubernetesProviderFactory.cs

```diff
@@ -1,41 +1,39 @@
 ﻿using System;
-
+
 using KubeClient;
-
+
 using Microsoft.Extensions.DependencyInjection;
-
+
 using Ocelot.Configuration;
 using Ocelot.Logging;
 using Ocelot.ServiceDiscovery;
+using Ocelot.ServiceDiscovery.Providers;
 
 namespace Ocelot.Provider.Kubernetes
 {
     public static class KubernetesProviderFactory
     {
-        public static ServiceDiscoveryFinderDelegate Get = (provider, config, route) =>
+        private const string PollKube = "pollkube";
+        public static ServiceDiscoveryFinderDelegate Get { get; } = CreateProvider;
+        private static IServiceDiscoveryProvider CreateProvider(IServiceProvider provider, ServiceProviderConfiguration config, DownstreamRoute route)
         {
             var factory = provider.GetService<IOcelotLoggerFactory>();
-            return GetKubeProvider(provider, config, route, factory);
-        };
-
-        private static ServiceDiscovery.Providers.IServiceDiscoveryProvider GetKubeProvider(IServiceProvider provider, ServiceProviderConfiguration config, DownstreamRoute route, IOcelotLoggerFactory factory)
-        {
             var kubeClient = provider.GetService<IKubeApiClient>();
 
-            var k8sRegistryConfiguration = new KubeRegistryConfiguration
+            var k8SRegistryConfiguration = new KubeRegistryConfiguration
             {
                 KeyOfServiceInK8s = route.ServiceName,
                 KubeNamespace = string.IsNullOrEmpty(route.ServiceNamespace) ? config.Namespace : route.ServiceNamespace,
             };
 
-            var k8sServiceDiscoveryProvider = new KubernetesServiceDiscoveryProvider(k8sRegistryConfiguration, factory, kubeClient);
+            var k8SServiceDiscoveryProvider = new KubernetesServiceDiscoveryProvider(k8SRegistryConfiguration, factory, kubeClient);
 
-            if (config.Type?.ToLower() == "pollkube")
+            if (config.Type?.ToLower() == PollKube)

```

Comment:
Why not?
```csharp
if (PollKube.Equals(config.Type, StringComparison.OrdinalIgnoreCase))
{
    // ... 
}
```

Please apply the same code snippet from PollConsul in Eureka too! We need to have a consistent design.